### ISSUE TITLE
Status of support for training on mobile

### ISSUE THREAD

**Issue by johanlantz** [pytorch/pytorch#38312]
Hi,

I need to build a prototype for federated learning on mobile (initially iOS). I have succeeded to transfer the pytorch model with torchscript and running inference of the dataset works perfectly in iOS.

Now for the next step I would need to actually train the model on the device and according to this:
https://github.com/pytorch/pytorch/pull/34958

it seems it might not yet be possible?

I did ask a similar question in November:
https://github.com/pytorch/pytorch/issues/29652
and got a reply from @dreiss that this was in progress but since that specific problem was different and 6 months have passed I closed the stale ticket and ask more concisely here instead if there is any new info.

What is the current status for training on mobile?

**Comment by johanlantz** [comment]
@ailzhang would it be possible to get any kind of indication if training should work on iOS or not? I am happy to continue testing but if it should not work then it would be a waste of time.

**Comment by ailzhang** [comment]
@johanlantz sorry I mistakenly label this issue as "triaged", it now should go into "mobile" triage queue and they should be able to answer your question. Sorry for the delay!

**Comment by johanlantz** [comment]
Thank you for the information, I look forward to the reply. The requirements are basic, just a simple neural net using the Salary dataset. I had this working with CoreML last year but we would like to it to torch so we have the same framework everywhere.

Just having a simple example of training on mobile (if it is supported) would be amazing.

**Comment by cereallarceny** [comment]
We have this completely working for federated learning on both Android and iOS if anyone is interested.

https://github.com/OpenMined/KotlinSyft
https://github.com/OpenMined/SwiftSyft

With that said, we're expecting to make our first public release and push to Maven and Cocoapods here in the next 2-3 weeks.

**Comment by ailzhang** [comment]
cc: @ljk53
